lib/spack/spack/detection/common.py

```diff
@@ -112,10 +112,15 @@ def path_to_dict(search_paths):
     # Reverse order of search directories so that a lib in the first
     # entry overrides later entries
     for search_path in reversed(search_paths):
-        for lib in os.listdir(search_path):
-            lib_path = os.path.join(search_path, lib)
-            if llnl.util.filesystem.is_readable_file(lib_path):
-                path_to_lib[lib_path] = lib
+        try:
+            for lib in os.listdir(search_path):
+                lib_path = os.path.join(search_path, lib)
+                if llnl.util.filesystem.is_readable_file(lib_path):
+                    path_to_lib[lib_path] = lib
+        except Exception as e:
```

Comment:
OSError?

Reply:
I think anything that is not an interrupt? The idea is that if something goes wrong we continue with the next directory. We can limit to OSError, but we might have to extend that in the future for the same reason as in #38733